test/e2e/deployment_test.go

```diff
@@ -31,12 +32,9 @@ func TestDeployment(t *testing.T) {
 	image := "gcr.io/knative-samples/helloworld-go"
 	_, err = test.WithServiceReady(ctx, "helloworld-go", testNamespace, image)
 	if err != nil {
-		t.Fatal("Knative Service not ready")
+		t.Fatal("Knative Service not ready", err)
 	}
 
-	// delete manually so that we can verify removing dependent operators
-	ctx.Cleanup()
```

Comment:
The cleanup was here on purpose so that the following function `WaitForOperatorDepsDeleted` can verify that all the dependencies that are installed automatically will be removed in the end. 
I suppose you're moving this to have a proper cleanup in the event of error? Maybe the Cleanup function should be idempotent so that we can call it multiple times - first in this place and second in the defer statement

Reply:
Good call, fixed